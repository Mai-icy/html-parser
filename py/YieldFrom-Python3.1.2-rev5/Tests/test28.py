#
#   Test throwing GeneratorExit into a subgenerator that
#   catches it and raises a different exception.
#

def f():
	try:
		print("Enter f")
		yield
		print("Exit f")
	except GeneratorExit:
		raise ValueError("Vorpal bunny encountered")

def g():
	print("Enter g")
	yield from f()
	print("Exit g")

gi = g()
next(gi)
gi.throw(GeneratorExit)
