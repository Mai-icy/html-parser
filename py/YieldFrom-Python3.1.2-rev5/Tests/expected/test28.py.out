Enter g
Enter f
Traceback (most recent call last):
  File "test28.py", line 9, in f
    yield
GeneratorExit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test28.py", line 21, in <module>
    gi.throw(GeneratorExit)
  File "test28.py", line 16, in g
    yield from f()
  File "test28.py", line 12, in f
    raise ValueError("Vorpal bunny encountered")
ValueError: Vorpal bunny encountered
